tests/mpi/data_out_faces_01.cc

```diff
@@ -0,0 +1,197 @@
+
+namespace pdd
+{
+  using namespace dealii;
+
+  // @sect3{The <code>PDDProblem</code> class template}
+
+  template <int dim>
+  class PDDProblem
+  {
+  public:
+    PDDProblem ();
+    ~PDDProblem ();
+    void run ();
+  private:
+    void setup_system ();
+    void output_results ();
+
+    MPI_Comm mpi_communicator;
+    parallel::distributed::Triangulation<dim>   triangulation;
+    //Triangulation<dim>   triangulation;
+    DoFHandler<dim>      dof_handler;
+    FE_Q<dim>            fe;
+
+    IndexSet             locally_owned_dofs;
+    IndexSet             locally_relevant_dofs;
+
+    ConstraintMatrix     constraints;
+
+    PETScWrappers::MPI::Vector locally_relevant_solution;
+  };
+
+  // The constructor and destructor of the class
+  template <int dim>
+  PDDProblem<dim>::PDDProblem () :
+    mpi_communicator (MPI_COMM_WORLD),
+    triangulation (mpi_communicator),
+    dof_handler (triangulation),
+    fe (1)
+  {}
+
+
+  template <int dim>
+  PDDProblem<dim>::~PDDProblem ()
+  {
+    dof_handler.clear ();
+  }
+
+  // @sect4{PDDProblem::setup_system}
+
+  template <int dim>
+  void PDDProblem<dim>::setup_system ()
+  {
+
+    // Initialize the mesh
+    std::vector<unsigned int> repetitions;
+    repetitions.push_back(10);
+    repetitions.push_back(12);
+
+    GridGenerator::subdivided_hyper_rectangle(triangulation,repetitions,Point<dim> (-1.,-1.),Point<dim> (1.,1.),true);
+    // triangulation.refine_global (1);
+
+    // Print out the mesh
+    dof_handler.distribute_dofs (fe);
+
+    // Initialize the solution
+    locally_owned_dofs = dof_handler.locally_owned_dofs ();
+    DoFTools::extract_locally_relevant_dofs (dof_handler,locally_relevant_dofs);
+    locally_relevant_solution.reinit(locally_owned_dofs,mpi_communicator);
+
+    locally_relevant_solution = 0;
+
+    // Apply a constant function at the boundary
+    constraints.clear ();
+    constraints.reinit (locally_relevant_dofs);
+    DoFTools::make_hanging_node_constraints (dof_handler,constraints);
+    VectorTools::interpolate_boundary_values (dof_handler, 0, ConstantFunction<dim> (1.0), constraints);
+    constraints.close ();
+    constraints.distribute(locally_relevant_solution);
+  }
+
+  // Generate the outputs
+  template <int dim>
+  void PDDProblem<dim>::output_results ()
+  {
+
+    // First generate an output for the cells
+    const unsigned int cycle = 1;
+
+    PETScWrappers::MPI::Vector solution (locally_owned_dofs,locally_relevant_dofs,mpi_communicator);
+    solution = locally_relevant_solution;
+
+    DataOut<dim> data_out;
+    data_out.attach_dof_handler (dof_handler);
+    data_out.add_data_vector (solution, "u");
+    Vector<float> subdomain (triangulation.n_active_cells());
+    for (unsigned int i=0; i<subdomain.size(); ++i)
+      subdomain(i) = triangulation.locally_owned_subdomain();
+    data_out.add_data_vector (subdomain, "subdomain");
+    // make sure the following works in parallel
+    data_out.build_patches ();
```

Comment:
Would you mind getting rid of the `subdomain` part? It is not related to the rest of the test.